crates/index/src/lib.rs

```diff
@@ -114,6 +123,10 @@ impl<O: Op> Index<O> {
         let options =
             serde_json::from_slice::<IndexOptions>(&std::fs::read(path.join("options")).unwrap())
                 .unwrap();
+        let flexible = serde_json::from_slice::<IndexFlexibleOptions>(
+            &std::fs::read(path.join("flexible")).unwrap_or_default(),
+        )
+        .unwrap_or_default();
```

Comment:
If the `setting` thread is killed before it writes the file. It lead to panics if it's opened after restarting.